lib/rouge/lexers/r.rb

```diff
@@ -68,7 +68,7 @@ def self.detect?(text)
         # highlighted specifically; thus use `Name::Function`.
         rule /\b(?<!.)(#{PRIMITIVE_FUNCTIONS.join('|')})(?=\()/, Name::Function
 
-        rule /[a-zA-Z.]([a-zA-Z_][\w.]*)?/ do |m|
+        rule /[a-zA-Z.]([a-zA-Z0-9_][\w.]*)?/ do |m|
```

Comment:
Now the character class is still missing `.`: for example, `..` is a valid identifier in R. And then, since `[a-zA-Z0-9_.]` is the same as `[\w.]`, might as well replace the whole expression with `[a-zA-Z.][\w.]*`.

Unfortunately this change will also wrongly accept `.1` as an identifier (luckily this is already included in the samples). The best way is probably therefore to treat these cases separately:

    /(?:[a-zA-Z]|\.[a-zA-Z_])[\w.]*/

This will ensure that an identifier either starts with a letter, or with a dot, followed by something other than a digit (i.e. a letter or underscore).

Reply:
I used a modified version of the regex listed in [this StackOverflow answer](https://stackoverflow.com/a/8396658/). This doesn't match `.` by itself (which I gather is a valid name?) and so I added that. The visual sample is all matching correctly now.